app/controllers/listings_controller.rb

```diff
@@ -34,9 +34,15 @@ def index
     respond_to do |format|
       # Keep format.html at top, as order is important for HTTP_ACCEPT headers with '*/*'
       format.html do
-        if request.xhr? && params[:person_id] # AJAX request to load on person's listings for profile view
-          @person = Person.find(params[:person_id])
-          PersonViewUtils.ensure_person_belongs_to_community!(@person, @current_community)
+
+        # Username is passed in person_id parameter for historical reasons.
+        # In the future the actual parameter name should be changed to better
+        # express its purpose. However this is a large change as there are quite a few
+        # resources nested under the people resource.
+        username = params[:person_id]
+
+        if request.xhr? && username # AJAX request to load on person's listings for profile view
+          @person = Person.find_by_username_and_community_id!(username, @current_community.id)
```

Comment:
Plan is to remove the method once we run the migrations and introduce the ``community_id`` field to person. Or at latest when the ``community_memberships`` table is dropped.